### PR TITLE
61-command-&-stops-minishell-running-properly

### LINKED ISSUE #61: `<command> &` stops minishell running properly.
background execution feature not implemented properly.

### PR DESCRIPTION
**Pull request by alex216**
fix #61 

## Summary by CodeRabbit

- **New Features**
	- Introduced handling for single ampersand (`&`) in the lexer, allowing for improved token processing.
	- Enhanced error handling for unexpected input in the execution process, providing clearer diagnostics.
- **Tests**
	- Added multiple test cases to validate the lexer’s handling of the ampersand character in various contexts, ensuring accurate token categorization.

### CODE REVIEW

File: src/lexer/constructor.c
Comment (rask24) [inline src/lexer/constructor.c:1]: no need to chage 🇩🇪

File: src/lexer/lexer_internal.h
Comment (rask24) [inline src/lexer/lexer_internal.h:1]: no need to change 🇪🇪

**Review by rask24 — COMMENTED** [review]
@alex216 The changes make sense and the lexer now properly handles '&'.
But I think it may be better to treat '&' as just a regular character, similar to how we handle '[' and '!'.
In bash, characters like '[' and '!' are regarded as operators, but in minishell we treat them as regular characters.
What do you think?

**Comment by alex216** [comment]
> @alex216 The changes make sense and the lexer now properly handles '&'. But I think it may be better to treat '&' as just a regular character, similar to how we handle '[' and '!'. In bash, characters like '[' and '!' are regarded as operators, but in minishell we treat them as regular characters. What do you think?

I agree with your proposal.

**Pull request merged**